leshan-server-core/src/main/java/org/eclipse/leshan/server/bootstrap/DefaultBootstrapSessionManager.java

```diff
@@ -68,24 +69,21 @@ public DefaultBootstrapSessionManager(BootstrapSecurityStore bsSecurityStore, Bo
      * @param securityChecker used to accept or refuse new {@link BootstrapSession}.
      */
     public DefaultBootstrapSessionManager(BootstrapSecurityStore bsSecurityStore, SecurityChecker securityChecker,
-            BootstrapTaskProvider tasksProvider, LwM2mBootstrapModelProvider modelProvider) {
+            BootstrapTaskProvider tasksProvider, LwM2mBootstrapModelProvider modelProvider,
+            BootstrapAuthorizer authorizer) {
```

Comment:
I think it would be better to remove `securityChecker`  and `bsSecurityStore` constructor argument as it should be just used to create an authorizer.

So better to force to pass an Authorizer (by checking is it not null)

By the way the 2 class attributes should be removed too.